### DISCUSSION TITLE
Looking to use JAMULUS for ADR practicing.

### DISCUSSION THREAD

**Discussion by Montxo** [jamulussoftware/jamulus#2383] [General]
Hello,
I am sending this message because I am trying to find a way to use JAMULUS together with an ONLINE  CLOUD application used for ADR Dialog Replacement education users.
Please allow me to explain about this more detailed.

I have developed an online CLOUD web site [www.idoblaje.com](http://www.idoblaje.com/) to allow the Users (teachers and students) rehearsal and record voices for films, etc. to users connected to this site.
I already have a working site for practicing recording and review for the dialogs. This is working well now.
For effective and good educational sessions the users need a realtime audio communication for rehearsal:

For example: 4 or 5 students and 1 instructor professor are connected at their own homes, each one of them connected to the [www.idoblaje.com](http://www.idoblaje.com/) site. We are currently using ZOOM for audio communication and sharing the professor display but…..
The audio latency for the students and very very high so the connected people cannot hear in “realtime” the audio that they need to while looking to the shared Instructor screen.
This is an capture of the current ONLINE application:

![idoblajeCLOUD](https://user-images.githubusercontent.com/913037/153721660-191e1d9d-7a25-40ac-8b87-ddce25a148ea.PNG)

 
So, I would try to find a way to add the low latency features of JAMULUS to provide AUDIO communication between the simultaneously user of my platform to practice and study DIALOG REPLACEMENT.

Anyone can let me know how can I do this or in fact maybe to hire a Jamulus programmer to add this task to my app or make an special version of JAMULUS for me?
I can give more details to someone interested to talk about this project.

Thanks a lot in advance. Take care.

Montxo Garcia
ZIO AUDIO
[www.zioaudio.com](http://www.zioaudio.com/)
SPAIN
whatsapp (+34) 607 293 186

**Comment by pljones** [comment]
@Montxo 

Jamulus is open source.  The developers are not paid for developing it.  You are free to pay someone to add features for you but your use is, of course, bound by the licence, which is GPL 2.0 (or later).

**Reply by Montxo** [reply]
My realtime WEB site for ADR practicing has an small cost to each user.
GPL can provide me with a way to use this open source as a plugin or additional use  together with my web site?
Jamulus should be a second application running. any solution for this? I dont know abot GPL too much.
Any help is appreciatted.

**Reply by gilgongo** [reply]
@Montxo When you say "Jamulus should be a second application running" then you are free to use it however you wish. If you wanted to **adapt** or change Jamulus's functionality in your application you can also do so. You (or your organisation) are free to make modifications to Jamulus and use them without ever releasing them. However, if you release changes **publicly** (and - copyright fans - here's where Jamulus is _unusually_ interesting!), the GPL requires you to make **all** the modified source code available to the public, under the GPL. [This FAQ](https://www.gnu.org/licenses/old-licenses/gpl-2.0-faq.en.html) may be helpful in answering other questions about what you can do.

BTW the reason I say Jamulus is unusually interesting is that @pljones said Jamulus is "bound by the licence, which is GPL 2.0 (or later)". ~However, the "or later" there is misleading because it's wrong :-)  You cannot expect to comply with provisions in the GPL **v3**. In particular,~ Jamulus can be run as either a client or a server from the same executable. This means its distribution under the GPL v2 _might not_ mean the same for application service providers as GPL v3. In contrast to the GPL v2 which is vague on this point, the GPL v3 states explicitly that there is no requirement to disclose the source code in an ASP use of a GPL program as long as a copy of the software is not sent to the client. 

We have discussed the possibility of moving Jamulus to the AGPL (which would be the ideal in Jamulus's case), but concluded it would be impractical.

**Reply by ghost** [reply]
Jamulus is licensed as (GPLv2 or later); meaning (GPL{version: 2,3,4,...}or later). This means that one can choose which legally existing license to use. One must then state which (GPLv{2,3,4,...}or later) license they are using and abide by it.

If the Jamulus developers move the license to a later version, then all code thereafter is under the later license. This means that if  one has a copy of the older version source code with its corresponding license, one is not affected by the developers new code and corresponding license unless one uses any of the new code.

If Jamulus has code which one is in legal compliance with but which prevents one from using a later version GPL and one wants to use a later version GPL, then modify the current  code to make it compliant with the later version GPL.

If Jamulus code is in non-compliance with GPLv2, Jamulus cannot be legally distributed until this bug is fixed.
If Jamulus code is in non-compliance with GPLv2 or later, Jamulus code may be legally modified by anyone to be in compliance with 'or later' because code cannot invalidate its license.

**Reply by Montxo** [reply]
> @Montxo When you say "Jamulus should be a second application running" then you are free to use it however you wish. If you wanted to **adapt** or change Jamulus's functionality in your application you can also do so. You (or your organisation) are free to make modifications to Jamulus and use them without ever releasing them. However, if you release changes **publicly** (and - copyright fans - here's where Jamulus is _unusually_ interesting!), the GPL requires you to make **all** the modified source code available to the public, under the GPL. [This FAQ](https://www.gnu.org/licenses/old-licenses/gpl-2.0-faq.en.html) may be helpful in answering other questions about what you can do.
> 
> BTW the reason I say Jamulus is unusually interesting is that @pljones said Jamulus is "bound by the licence, which is GPL 2.0 (or later)". ~However, the "or later" there is misleading because it's wrong :-) You cannot expect to comply with provisions in the GPL **v3**. In particular,~ Jamulus can be run as either a client or a server from the same executable. This means its distribution under the GPL v2 _might not_ mean the same for application service providers as GPL v3. In contrast to the GPL v2 which is vague on this point, the GPL v3 states explicitly that there is no requirement to disclose the source code in an ASP use of a GPL program as long as a copy of the software is not sent to the client.
> 
> We have discussed the possibility of moving Jamulus to the AGPL (which would be the ideal in Jamulus's case), but concluded it would be impractical.

Thanks a lot for the explanation GILGONGO.

**Reply by Montxo** [reply]
> Jamulus is licensed as (GPLv2 or later); meaning (GPL{version: 2,3,4,...}or later). This means that one can choose which legally existing license to use. One must then state which (GPLv{2,3,4,...}or later) license they are using and abide by it.
> 
> If the Jamulus developers move the license to a later version, then all code thereafter is under the later license. This means that if one has a copy of the older version source code with its corresponding license, one is not affected by the developers new code and corresponding license unless one uses any of the new code.
> 
> If Jamulus has code which one is in legal compliance with but which prevents one from using a later version GPL and one wants to use a later version GPL, then modify the current code to make it compliant with the later version GPL.
> 
> If Jamulus code is in non-compliance with GPLv2, Jamulus cannot be legally distributed until this bug is fixed. If Jamulus code is in non-compliance with GPLv2 or later, Jamulus code may be legally modified by anyone to be in compliance with 'or later' because code cannot invalidate its license.

THANKS DavidSavinkoff for the help.

**Reply by gilgongo** [reply]
@DavidSavinkoff ~I'm afraid Jamulus is distributed [exclusively under the terms of  GPL v2](https://github.com/jamulussoftware/jamulus/blob/master/COPYING) - we do not as far as I am aware say "or any later version" :-( So we're just like the Linux kernel :-)~

We DO plan to have an exception statement to allow distribution from the Apple Appstore though, but that's a bit of a different matter.

**Reply by hoffie** [reply]
> @DavidSavinkoff I'm afraid Jamulus is distributed [exclusively under the terms of GPL v2](https://github.com/jamulussoftware/jamulus/blob/master/COPYING) - we do not as far as I am aware say "or any later version" :-( So we're just like the Linux kernel :-)

Are you sure? The linked file is a copy of GPLv2 and I assume that it will always be that.

The code -- and I think this is what matters -- explicitly mentions [This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version.](https://github.com/jamulussoftware/jamulus/blob/master/src/main.cpp#L11-L12)

**Reply by gilgongo** [reply]
@hoffie Ah - so it does! :-) In which case, the question is has it always said that? In which case we're OK (although the "ASP loophole" we discussed still exists).

**Reply by hoffie** [reply]
> In which case, the question is has it always said that?

The initial version commit (f787f8dd5) already includes it, so it has been there for a very long time. Can't verify for the pre-git versions obviously.

**Reply by gilgongo** [reply]
That's good then. I guess I must be used to seeing distributor preambles in licence files under which the licence is then pasted, so Github fooled me into thinking we were GPL v2 only. I don't think it changes anything though.

**Comment by softins** [comment]
I'm not sure that Jamulus would necessarily help. The purpose of the low latency in Jamulus is so that the participants in a group can hear each other approximately at the same time, which is important for a group of musicians playing together.

It appears that the requirement above is to synchronise the audio with a video stream. A video stream will have its own latency, which might be different for different participants watching the same stream. Jamulus would not necessarily help with that.

Or have I misunderstood?

**Reply by Montxo** [reply]
Hello. The sync between video and audio is already resolved in my web appilcation. This already record and replay in sync the audio with video.
What I am looking for is, exactly what you mention: ACTORS (students) and like the musicians. They need to hear each other "ACTORS" playing his role at the same time while rehearsing before they record their roles in my web site.

It is exactly like the musicians requirements. 
Just hear each other at the same time to play in sync.

Sorry if my english is not too good to explain accurately.
Hope this help you to understand.

**Reply by dcorson-ticino-com** [reply]
I don't exactly understand your requirements, but running Zoom and Jamulus at the same time is very possible and used, for example, for the WorldJam concerts.
Just turn the audio in Zoom off and use Jamulus for the audio.  You get low latency sound of Jamulus and the Zoom typical high latency video.

**Reply by Montxo** [reply]
Hello. thanks for the reply.
We will use ZOOM to share the screen of ONE of the participants. It seems that when sharing the screen the audio transmision has a better lower latency... but audio of the rest of the participants (not sharing screen) has a high latency. So this is the reason we need to find a way to provide low latency to the NOT SHARING PARTICIPANTS.

**Comment by chrisrimple** [comment]
It seems like Jamulus, without modification, may provide you with what you're seeking: multiple students able to hear each other (and the instructor) in "real time" and respond.  This the same for musicians and choral singers using Jamulus.

BUT, the latency that students experience will depend on their physical distance from the Jamulus server.  If you want near-real-time sound, everyone should be within 500 miles of the Jamulus server.

**Reply by Montxo** [reply]
Hi Chrisrimple. thansk for the reply.
Then the critical issue it to set a JAMULUS server within those 500 miles. Can anybody help me on how can I do this? For example, I would need a server in MADRID SPAIN at the moment. 
Can anybody help for this? THANKS.

**Reply by ignotus666** [reply]
Have you had a look at the [website](https://jamulus.io/es/)? Está en castellano también.
